### PR TITLE
Add OKD compatibility to crc for building CRC for OKD bundles

### PR DESCRIPTION
**Pull request by cgruver**
**Fixes:** Issue #977 

**Relates to:** https://github.com/code-ready/snc/issues/212

## Solution/Idea

I modified Makefile, constants.go, and version.go to allow for building a `crc` image which uses OKD instead of OCP.  The building of `crc` should not be affected by these changes, as it is the default.   `crc` for OKD can be built by setting environment variables.

## Proposed changes

1. Removed `DefaultOcURLBase` from `constants.go` and added it as a build config variable to `version.go`
1. Modified the creation of `ocURLForOs` to account for okd vs. ocp URLs
1. Modified `Makefile` to allow overriding `BUNDLE_VERSION`
1. Modified `Makefile` to add `MIRROR` as an override for `version.defaultOcURLBase` and added `https://mirror.openshift.com/pub/openshift-v4/clients/ocp` as a default value.

## Testing

I have tested the OKD builds of snc and crc.  I do not have the facility to test OCP builds.

To build for OKD:

Build `snc`:  See Pull Request - https://github.com/code-ready/snc/pull/230

```bash
export BUNDLE_VERSION=${OPENSHIFT_VERSION}                                  # Using OPENSHIFT_VERSION that was set for snc.
export MIRROR=https://github.com/openshift/okd/releases/download     
export BUNDLE_DIR=/tmp/snc                                                                      # where snc was built in /tmp/snc

make embed_bundle
```

Use `{"auths":{"fake":{"auth": "Zm9vOmJhcgo="}}}` as your pull secret when prompted during `crc start`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -159,7 +161,7 @@ cross-lint: $(GOPATH)/bin/golangci-lint
 gen_release_info:
 	@cat release-info.json.sample | sed s/@CRC_VERSION@/\"$(CRC_VERSION)\"/ > $(RELEASE_INFO)
 	@sed -i s/@GIT_COMMIT_SHA@/\"$(COMMIT_SHA)\"/ $(RELEASE_INFO)
-	@sed -i s/@OPENSHIFT_VERSION@/\"$(BUNDLE_VERSION)\"/ $(RELEASE_INFO)
+	@sed -i s/@BUNDLE_VERSION@/\"$(BUNDLE_VERSION)\"/ $(RELEASE_INFO)
```
Comment (anjannath) [inline Makefile:164]: This is not needed i think, we only have `@OPENSHIFT_VERSION@` in that file

**Comment by anjannath** [comment]
@cgruver The first three commits and the last two commits can be squashed, so there will be two commits, or maybe it makes sense to have it all in one commit, now i see changes from previous commits are reverted in later commits.

**Review by cfergeau — COMMENTED** [review]
Nit: This PR is not "adding OKD compatibility", it's more about adding knobs which can be changed externally in order to do OKD builds. If you don't know what these knobs should be set to, no OKD build for you. Maybe this can be done through `make okd`, or maybe this will need a `BUNDLE_TYPE` Makefile variable which will be used in the targets which need it. Or maybe this should be dynamic, with a single crc binary being able to support both bundles?

You'll need to cleanup the git history too, probably by squashing everything together, and adding a bit of background on what you did, https://www.berrange.com/posts/2012/06/27/thoughts-on-improving-openstack-git-commit-practicehistory/

**Comment by cgruver** [comment]
I will clean up the commits and re-submit this.  @cfergeau Thanks for the tips!

**Comment by cgruver** [comment]
Submitted clean request: https://github.com/code-ready/crc/pull/1511

**Pull request closed without merging**